Battleships/Battleships/BoardManager.cs

```diff
@@ -0,0 +1,306 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Battleships
+{
+    public struct Cell
+    {
+        public bool hasShip;
+        public bool isHit;
+        public int shipType;
+
+        public Cell(bool hasShip, int shipType)
+        {
+            this.hasShip = hasShip;
+            this.shipType = shipType;
+            isHit = false;
+        }
+    }
+    public class BoardManager
+    {
+        private int _height;
+        private int _width;
+        private int _placedShips;
+        private Cell[][] board;
+
+        public Cell[][] Board => board;
+        public int PlacedShips => _placedShips;
+
+        public BoardManager(int height, int width)
+        {
+            this._height = height;
+            this._width = width;
+            _placedShips = 0;
+            board = new Cell[10][];
+
+            for (int i = 0; i < height; i++)
+            {
+                board[i] = Enumerable.Repeat(new Cell(false, 0), width).ToArray();
+            }
+        }
+
+        public void PlaceShip(int i, int j, Target ship)
+        {
+            TryPlaceShip(i, j, ship);
+        }
+
+        public bool HasShip(int i, int j)
+        {
+            if (CheckPlacementBounds(i, j))
+            {
+                return board[i][j].hasShip;
+            }
+            throw new InvalidOperationException("Out of bounds");
+        }
+
+        public bool IsHit(int i, int j)
+        {
+            if (CheckPlacementBounds(i, j))
+            {
+                return board[i][j].isHit;
+            }
+            throw new InvalidOperationException("Out of bounds");
+        }
+
+        public void HitSquare(int i, int j)
+        {
+            if (CheckPlacementBounds(i, j) && !IsHit(i, j))
+            {
+                board[i][j].isHit = true;
+            }
+        }
+
+
+        public void RandomPlaceShip(int count, Target ship)
+        {
+            if (count > 100) throw new InvalidOperationException("Count can not exceed total cell count, total cell count = 100!");
+            Random random = new Random(10);
+
+            int shipCounter = 0;
+
+            while (shipCounter < count)
+            {
+                if (TryPlaceShip(random.Next(0, _height), random.Next(0, _width), ship))
+                {
+                    shipCounter++;
+
+                }
+            }
+        }
+
+        public bool TryPlaceShip(int x, int y, Target ship)
+        {
+            if (ship.Size == 1)
+            {
+                if (CheckPlacementBounds(x, y, ship))
+                {
+                    if (HasShip(x, y) || !CheckAdjacentSquares(x, y))
+                    {
+                        return false;
+                    }
```

Comment:
Tek satıra alalım

Reply:
Tek satır yapabildiğin yerler varsa bu tarz onlara bakıp yaparsan güzel olur ben sadece 1 tanesini gösterdim tüm projeyi kapsıyor.